rtcp.c

```diff
@@ -327,6 +348,113 @@ int janus_rtcp_fix_ssrc(char *packet, int len, int fixssrc, uint32_t newssrcl, u
 	return 0;
 }
 
+
+int janus_rtcp_process_incoming_rtp(rtcp_context *ctx, char *packet, int len, int max_nack_queue) {
+	if(ctx == NULL || packet == NULL || len < 1)
+		return -1;
+
+	ctx->enabled = 1;
+	/* Parse this RTP packet header and update the rtcp_context instance */
+	rtp_header *rtp = (rtp_header *)packet;
+	uint16_t seq_number = ntohs(rtp->seq_number);
+	if(ctx->base_seq == 0 && ctx->seq_cycle == 0)
+		ctx->base_seq = seq_number;
+
+	if(max_nack_queue + seq_number < ctx->last_seq_nr)
+		ctx->seq_cycle++;
+	ctx->last_seq_nr = seq_number;
+	ctx->received++;
+	uint32_t rtp_expected = 0x0;
+	if(ctx->seq_cycle > 0) {
+		rtp_expected = ctx->seq_cycle;
+		rtp_expected = rtp_expected << 16;
+	}
+	rtp_expected = rtp_expected + 1 + seq_number - ctx->base_seq;
+	ctx->lost = rtp_expected - ctx->received;
+	ctx->expected = rtp_expected;
+
+	uint64_t arrival = (janus_get_monotonic_time() * ctx->tb) / 1000000;
+	uint64_t transit = arrival - ntohl(rtp->timestamp);
+	uint64_t d = abs(transit - ctx->transit);
+	ctx->transit = transit;
+	ctx->jitter += (1./16.) * ((double)d  - ctx->jitter);
+
+	return 0;
+}
+
+
+uint32_t janus_rtcp_context_get_lsr(rtcp_context *ctx) {
+	return ctx ? ctx->lsr : 0;
+}
+
+uint32_t janus_rtcp_context_get_received(rtcp_context *ctx) {
+	return ctx ? ctx->received : 0;
+}
+
+uint32_t janus_rtcp_context_get_lost(rtcp_context *ctx) {
+	if(ctx == NULL)
+		return 0;
+	uint32_t lost;
+	if(ctx->lost > 0x7fffff) {
+		lost = 0x7fffff;
+	} else {
+		lost = 0x00FFFFFF & ctx->lost;
```

Comment:
you mix upper-case and lower-case hex literals here. also, the mask in the else case is redundant - there can't be any high bits set to mask off if `! (ctx->lost > 0x7FFFFF)`